### DISCUSSION TITLE
能否出个WPF样式界面的高级教程

### DISCUSSION THREAD

**Discussion by cpydown** [BYJRK/DotNet-Discussions#34] [WPF 开发]
工作上得按照美工设计的效果图来实现，而往往现有的控件满足不了，就需要定制控件样式，这方面网上资料比较少，都是简单提两句样式就完了，没有深入

**Comment by BYJRK** [comment]
你可以举个例子，比如说你现在想要实现什么样式的什么控件。定制样式涉及到的知识太多，没办法简单覆盖。

**Comment by cpydown** [comment]
以Expander为例

类似前端ElementUI中侧向导航NavMenu（[链接](https://element.eleme.cn/#/zh-CN/component/menu)），可以多级展开的效果，我用Expander尝试修改，不是很成功，而HandyControl、MaterialDesignToolkit等第三方控件的效果要么和效果图差别比较大，要么只支持一级展开

另外有时需要仿一些古老软件的界面，比如[这里](https://www.51wendang.com/doc/fe158d4f6164e2d183e8bb5a)的图片中XP系统左侧的“控制面板”、“请参阅”的Expander，怎样通过定制Expander的样式来满足？

**Reply by Hakoyu** [reply]
这个很简单啊 在`Expander`塞`ListBox`就好了

**Reply by cpydown** [reply]
> 这个很简单啊 在`Expander`塞`ListBox`就好了

主要是样式不好调，把Expander的控件模板样式取出来后，不知道怎么修改，比如ContentPresenter, TemplateBinding等怎么设置

**Comment by cpydown** [comment]
这两天看了Up的原神角色那个视频，大佬能把P3里美化界面用到的样式技术给讲讲就挺好
